PXB-3162 Describe the options available for xtrabackup redo logs (8.0) (#181)
percona/pxb-docs: docs/redo-log-archiving.md

# InnoDB redo log archiving

> A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represent the data in the redo log. 

Reviewer: ```suggestion
A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represents the data in the redo log. 
```

> Data in the redo log is appended and the movement of the checkpoint truncates old data. 

Reviewer: ```suggestion
Data in the redo log is appended, and the movement of the checkpoint truncates old data. 
```

> If the MySQL server has significant activity during a backup operation and the redo log file storage media operates faster than the backup storage media, the backup operation may fail to keep pace with the redo log generation and cause a loss of data.

Reviewer: ```suggestion
If the MySQL server has significant activity during a backup operation and the redo log file storage media operates faster than the backup storage media, the backup operation may fail to keep pace with the redo log generation and cause data loss.
```

Prior to the redo log archiving feature, the solutions were:

* Increasing the redo log size

* Checking for I/O congestion if the read speed is slow

* Checking for I/O or network congestion if the write speed is slow

* Taking the backup at an off-peak time

## redo log archiving

> MySQL 8.0.17 added redo log archiving. To enable this feature, the follow settings are required:

Reviewer: ```suggestion
MySQL 8.0.17 added redo log archiving. To enable this feature, the following settings are required:
```

    | innodb_redo_log_archive_dirs | backup1:/var/lib/mysql-redo-archive/ |
    +------------------------------+--------------------------------------+
    1 row in set (0.0025 sec)
    ```

Users with the [INNODB_REDO_LOG_ARCHIVE] privilege can call the `innodb_redo_log_archive_start()` or `innodb_redo_log_archive_stop()` functions. to either start or stop archiving. 

```{.bash data-prompt="mysql>"}
mysql> SELECT innodb_redo_log_archive_start('backup1','backup1');
```

??? example "Expected output"
    ```{.text .no-copy}
    +-----------------------------------------------------+
    | innodb_redo_log_archive_start('backup1','backup1')  |
    +------------------------------+----------------------+
    |                                                   0 |
    +------------------- ---------------------------------+
    1 row in set (0.00225 sec)
    ```

## Percona XtraBackup and redo log archiving

Percona XtraBackup 8.0.34-29 added the [--redo-log-arch-dir](xtrabackup-option-reference.md#-redo-log-arch-dirname) option. This option lets you configure a redo log archiving directory, if this directory is not already configured in MySQL.

> To use this option you must run Percona XtraBackup as the owner of `mysqld`. If Percona XtraBackup does not have the appropriate permissions, then the action generates a "Permission denied" error and archiving is not used. Percona XtraBackup creates a temporary directory for the archive file.

Reviewer: ```suggestion
To use this option, you must run Percona XtraBackup as the owner of `mysqld`. If Percona XtraBackup does not have the appropriate permissions, then the action generates a "Permission denied" error, and archiving is not used. Percona XtraBackup creates a temporary directory for the archive file.
```

# InnoDB redo log archiving

> A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represents the data in the redo log.

Reviewer 2: ```suggestion
A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains entries necessary to bring the database back to a consistent state.
```

> Data in the redo log is appended, and the movement of the checkpoint truncates old data.

Reviewer 2: ```suggestion
Data in the redo log is always appended. When a file is full, a new file is created and the movement of the checkpoint truncates old files.
```

```{.bash data-prompt="mysql>"}
mysql> SELECT innodb_redo_log_archive_start('backup1','backup1');
```

??? example "Expected output"
    ```{.text .no-copy}
    +-----------------------------------------------------+
    | innodb_redo_log_archive_start('backup1','backup1')  |
    +------------------------------+----------------------+
    |                                                   0 |
    +------------------- ---------------------------------+
    1 row in set (0.00225 sec)
    ```

## Percona XtraBackup and redo log archiving

Percona XtraBackup 8.0.34-29 added the [redo-log-arch-dir](xtrabackup-option-reference.md#-redo-log-arch-dirname) option. If the archiving directory is not configured in MySQL, Percona XtraBackup creates a temporary directory for the archive file. To use this option, you must run Percona XtraBackup as the owner of `mysqld`. If Percona XtraBackup does not have the appropriate permissions, then the action generates a "Permission denied" error, and archiving is not used.

```{.bash data-prompt}
$ sudo -H -u mysql bash

> $ xtrabackup --no-lock=1 --compress --parallel=4 --host=localhost --user=root --password='password_string' --backup=1 --target-dir=/Backup/13oct 2> /tmp/b0-with-redo-archiving-as-mysql-os-user.log

Reviewer 2: Where is the `--redo-log-arch-dir` parameter been set?

Author: I don't understand the question.
